hat/assets/js/apps/Iaso/domains/workflows/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FunctionComponent, useMemo } from 'react';
+// @ts-ignore
+import { makeStyles, Box } from '@material-ui/core';
+// @ts-ignore
+import { commonStyles, useSafeIntl } from 'bluesquare-components';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { baseUrls } from '../../constants/urls';
+
+import TopBar from '../../components/nav/TopBarComponent';
+import { TableWithDeepLink } from '../../components/tables/TableWithDeepLink';
+import { Filters } from './components/Filters';
+
+import { useGetWorkflowVersions } from './hooks/requests/useGetWorkflowVersions';
+import { useGetType } from '../entities/entityTypes/hooks/requests/entitiyTypes';
+import { WorkflowsParams } from './types/workflows';
+
+import { redirectToReplace } from '../../routing/actions';
+
+import MESSAGES from './messages';
+import { useGetColumns, defaultSorted, baseUrl } from './config';
+
+const useStyles = makeStyles(theme => ({
+    ...commonStyles(theme),
+}));
+// TODO: centralize those types
+type RouterCustom = {
+    prevPathname: string | undefined;
+};
+type State = {
+    routerCustom: RouterCustom;
+};
+type Router = {
+    goBack: () => void;
+};
+type Props = {
+    params: WorkflowsParams;
+    router: Router;
+};
+
+export const Workflows: FunctionComponent<Props> = ({ params, router }) => {
+    const prevPathname = useSelector(
+        (state: State) => state.routerCustom.prevPathname,
+    );
+    const dispatch = useDispatch();
+    const classes: Record<string, string> = useStyles();
+    const { formatMessage } = useSafeIntl();
+    const { data, isFetching } = useGetWorkflowVersions(params);
+    const { entityTypeId } = params;
+    const columns = useGetColumns(entityTypeId);
+
+    const { data: entityType } = useGetType(entityTypeId);
+    const title = useMemo(
+        () =>
+            entityType
+                ? `${entityType.name} - ${formatMessage(MESSAGES.workflows)}`
+                : '',
+        [entityType, formatMessage],
+    );
+    return (
+        <>
+            <TopBar
+                title={title}
+                displayBackButton
+                goBack={() => {
+                    if (prevPathname) {
+                        router.goBack();
+                    } else {
+                        dispatch(redirectToReplace(baseUrls.entityTypes, {}));
+                    }
+                }}
```

Comment:
We could make a hook for this. It would save us the import and declaration of `dispatch` and `prevPathName`